src/main/java/ch/njol/skript/expressions/ExprLastDeathTime.java

```diff
@@ -0,0 +1,181 @@
+		return null;
+	}
+
+	/*
+	 * Date#compare(date)
+	 * This method returns whether the passed date is
+	 * before, the same as, or after the date it's being compared to.
+	 * A value less than 0 indicates that the passed date is AFTER the date it's being compared to.
+	 * A value of 0 indicates that the passed date is the SAME the date it's being compared to.
+	 * A value greater than 0 indicates that passed date is BEFORE the date it's being compared to.
+	 * 
+	 * We use this method because we don't want the new death date to be in the future
+	 */
+	@SuppressWarnings("null")
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+
+		if (delta == null)
+			return;
+
+		Date now = Date.now();
+
+		for (Player player : getExpr().getArray(e)) {
+			if (mode == ChangeMode.ADD) {
+
+				// Get the current death time of the player.
+				Date deathTime = convert(player);
+
+				deathTime.add((Timespan) delta[0]);
+
+				if (deathTime.compareTo(now) < 1) {
+					// Get the timespan representing the new time since death (in ticks).
+					long newTimespanInTicks= deathTime.difference(now).getTicks_i();
+
+					/*
+					 *  Convert to int.
+					 *  If it is greater than the max value for an integer,
+					 *  the user probably wants to set it to the max value, right?
+					 */
+					int newTimeSinceDeath = newTimespanInTicks < Integer.MAX_VALUE ? (int) newTimespanInTicks : Integer.MAX_VALUE;
+
+					player.setStatistic(Statistic.TIME_SINCE_DEATH, newTimeSinceDeath);
+				}
+
+			} else if (mode == ChangeMode.REMOVE) {
+
+				// Get the current death time of the player.
+				Date deathTime = convert(player);
+
+				deathTime.subtract((Timespan) delta[0]);
+				
+				// Get the timespan representing the new time since death (in ticks).
+				long newTimespanInTicks= deathTime.difference(now).getTicks_i();
+
+				/*
+				 *  Convert to int.
+				 *  If it is greater than the max value for an integer,
+				 *  the user probably wants to set it to the max value, right?
+				 */
+				int newTimeSinceDeath = newTimespanInTicks < Integer.MAX_VALUE ? (int) newTimespanInTicks : Integer.MAX_VALUE;
+
+				player.setStatistic(Statistic.TIME_SINCE_DEATH, newTimeSinceDeath);
+
+			} else if (mode == ChangeMode.SET) {
+
+				/*
+				 * Since the statistic is actually the time since the last death of the player,
+				 * it needs to be set to the difference between now and the new date.
+				 * For example, if we were setting the player's last death to 1 day ago,
+				 * the statistic would need to be 24 hours, and that's what this would give us.
+				 * 
+				 * If the new date is in the future, the statistic will remain unchanged.
+				 */
+
+				Date newDate = ((Date) delta[0]);
+
+				if (newDate.compareTo(now) < 1) {
+					long newTimespanInTicks = ((Date) delta[0]).difference(now).getTicks_i();
+					int newTimeSinceDeath = newTimespanInTicks < Integer.MAX_VALUE ? (int) newTimespanInTicks : Integer.MAX_VALUE;
```

Comment:
you could also do `int newTimeSinceDeath = Math.min(newTimespanInTicks, Integer.MAX_VALUE);`